lib/gssapi/krb5/acquire_cred.c

```diff
@@ -80,6 +80,34 @@ get_keytab(krb5_context context, krb5_keytab *keytab)
     return (kret);
 }
 
+static krb5_error_code
+get_client_keytab(krb5_context context,
+		  krb5_const_principal principal,
+		  krb5_keytab *keytab)
+{
+    krb5_error_code kret;
+    krb5_keytab_entry entry;
+    char *name = NULL;
+
+    kret = _krb5_kt_client_default_name(context, &name);
```

Comment:
Not a required change, but you can deindent a lot like so:

```
    kret = _krb5_kt_client_default_name(context, &name);
    if (kret == 0)
        kret = krb5_kt_resolve(context, name, keytab);
    if (kret == 0 && principal != NULL)
	kret = krb5_kt_get_entry(context, *keytab, principal,
				 0, 0, &entry);
    if (kret == 0)
	krb5_kt_free_entry(context, &entry);
    krb5_xfree(name);
    if (kret == 0)
        return 0;

    return get_keytab(context, keytab); /* fall back on regular keytab */
```

Reply:
Yeah, thought about that, wasn't convinced it was necessarily more readable. But happy to.